Update dolt-sql-procedures.md (#2532)
dolthub/docs: packages/dolt/content/reference/sql/version-control/dolt-sql-procedures.md

## `dolt_stats_restart()`

If no thread is active for the current database, start a new update
thread with the current session's parameters (`dolt_stats_memory_only`, `dolt_stats_job_interval`, `dolt_stats_gc_enabled`).
If a thread is already active for
this database, the thread is stopped and started with the new
parameters.

## `dolt_stats_stop()`

> Clear the work queue and stop the thread

Reviewer: ```suggestion
Clears the work queue and stops the thread
```

(otherwise no-op).

## `dolt_stats_purge()`

> Delete the stats cache from memory and the filesystem. Also clearing working queue and stop the update thread.

Reviewer: ```suggestion
Deletes the stats cache from memory and the filesystem. Also clearing working queue and stop the update thread.
```

## `dolt_stats_once()`:

> This command cillect statistics once. It should be used when no background thread is running

Reviewer: ```suggestion
This command collects statistics once. It should be used when no background thread is running
```

(ex: in `dolt sql` and when we do not wish to run a background thread).

## `dolt_stats_wait()`:

> Block on a full queue cycle. In practice it takes at least one cycle for stats to reflect the contents

Reviewer: ```suggestion
Blocks on a full queue cycle. In practice it takes at least one cycle for stats to reflect the contents
```

of the database stats in the blocking session.

## `dolt_stats_gc()`:

Blocks waiting for a GC signal. Garbage collection finalizes in the same cadence as new statistic updates.

## `dolt_stats_flush()`:

> Block waiting on a flush signl. Flushes occur after new statistic updates.

Reviewer: ```suggestion
Blocks waiting on a flush signal. Flushes occur after new statistic updates.
```

## `dolt_stats_info()`:

> Print the current state of the stats provider (optional `'-short'` flag).

Reviewer: ```suggestion
Returns the current state of the stats provider (optional `'-short'` flag).
```
